### ISSUE TITLE
array_map() ignores ignoreInternalFunctionFalseReturn

### ISSUE THREAD

**Issue by brewers-dev-alex** [vimeo/psalm#2727]
I am using array_map() to map an array that contains the result of DateTime::format() to a function that expects a string. DateTime::format() can return false on error, but I expect the ignoreInternalFunctionFalseReturn flag to prevent psalm from raising any errors, but it still reports InvalidScalarArgument.

Example code here:

https://psalm.dev/r/ce15bc47ac
```php
<?php

$date = new DateTime();

$a = [$date->format('Y-m-d')];

array_map('takesString', $a);
takesString($a[0]);
takesString($date->format('Y-m-d'));
foreach($a as $string) {
  takesString($string);
}

/**
 * @param string $string
 * @return string
 */
function takesString(string $string): string {
    return $string;
}
```

**Comment by brewers-dev-alex** [comment]
Thanks for fixing this so quickly.
